### PR TITLE
[C-2579][C-2580] audius-query

### PR DESCRIPTION
**Pull request by amendelsohn**
### Description

Simple tool to create all-in-one slice, fetch, and select hooks that integrate automatically with audius-client's existing cache system.

basic usage:
- `createApi` with a list of endpoints
- `use<ENDPOINT_NAME>` hook for each endpoint is automatically exported for use in components

Future work:
- complete apiResponseSchema
- add options for disabling fetch, pagination, etc
- type variables for args and return type so hook usages are type safe

Covers basic use cases for now.

Usage docs to come later...

### Dragons

This will sit alongside existing patterns, but the goal should be a complete migration

### How Has This Been Tested?

Local dev (web)

### CODE REVIEW

File: packages/common/src/api/types.ts [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  schemaKey?: string
+  kind?: Kind
+}
+
+export type EndpointConfig = {
+  fetch: (fetchArgs: any, context: AudiusQueryContextType) => Promise<any>
+  options?: EndpointOptions
+}
+
+export type EntityMap = {
```
Comment (dylanjeffers) [inline packages/common/src/api/types.ts:32]: can possibly improve this type

**Review by dylanjeffers — APPROVED** [review]
send itt! this this amazing

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/audius-query

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/audius-query

**Pull request merged**